app/controllers/api/v2/registration_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Api
+  module V2
+    class RegistrationController < V2::BaseController
+      include Api::Version2
+      include Foreman::Controller::Registration
+      include Foreman::Controller::Parameters::Host
+
+      rescue_from ActiveRecord::RecordNotFound do |error|
+        logger.info "#{error.message} (#{error.class})"
+        not_found(error.message)
+      end
+
+      api :GET, '/register', N_('Render Global registration template')
+      param :organization_id, :number, desc: N_("ID of the Organization to register the host in.")
```

Comment:
How do we validate this data, e.g. how do we make sure I don't register a host in an organization I'm not allowed to?  Shouldn't we bake this information into the JWT token?

Reply:
This is not an issue, user is already authenticated by that time, so regular access is enforce, I double-checked that

```
curl -X GET "https://luna-nightly.example.tst/register?organization_id=3&location_id=2&operatingsystem_id=3" --user manager:changeme
echo "ERROR: not_found";
echo "Organization with id 3 not found";
exit 1
```

however it exposed the other method problem. Baking of that in JWT prevents that, I'll add it to the comment below.